algorithms/benches/snark/marlin.rs

```diff
@@ -100,45 +64,141 @@ fn snark_circuit_setup(c: &mut Criterion) {
 fn snark_prove(c: &mut Criterion) {
     c.bench_function("snark_prove", move |b| {
         let num_constraints = 100;
-        let num_variables = 100;
+        let num_variables = 25;
+        let mul_depth = 1;
         let rng = &mut TestRng::default();
 
-        let x = Fr::rand(rng);
-        let y = Fr::rand(rng);
-
         let max_degree = AHPForR1CS::<Fr, MarlinHidingMode>::max_degree(1000, 1000, 1000).unwrap();
         let universal_srs = MarlinInst::universal_setup(&max_degree).unwrap();
         let fs_parameters = FS::sample_parameters();
 
-        let circuit = Benchmark::<Fr> { a: Some(x), b: Some(y), num_constraints, num_variables };
+        let (circuit, _) = TestCircuit::gen_rand(mul_depth, num_constraints, num_variables, rng);
 
         let params = MarlinInst::circuit_setup(&universal_srs, &circuit).unwrap();
         b.iter(|| MarlinInst::prove(&fs_parameters, &params.0, &circuit, rng).unwrap())
     });
 }
 
+fn snark_batch_prove(c: &mut Criterion) {
+    c.bench_function("snark_batch_prove", move |b| {
+        let num_constraints_base = 100;
+        let num_variables_base = 25;
+        let mul_depth_base = 1;
+        let rng = &mut TestRng::default();
+
+        let max_degree = AHPForR1CS::<Fr, MarlinHidingMode>::max_degree(1000000, 1000000, 1000000).unwrap();
+        let universal_srs = MarlinInst::universal_setup(&max_degree).unwrap();
+        let fs_parameters = FS::sample_parameters();
+
+        let circuit_batch_size = 5;
+        let instance_batch_size = 5;
+
+        let mut pks = Vec::with_capacity(circuit_batch_size);
+        let mut all_circuits = Vec::with_capacity(circuit_batch_size);
+        let mut keys_to_constraints = BTreeMap::new();
+        for i in 0..circuit_batch_size {
+            let num_constraints = num_constraints_base + i;
+            let num_variables = num_variables_base + i;
+            let mul_depth = mul_depth_base + i;
+            // if i > 0 {
+            //     num_constraints += 5000;
+            //     num_variables += 5000;
+            //     mul_depth += 5000;
+            // }
```

Comment:
Is this intended to be commented out?

Reply:
It was, but you're right for calling it out, removing it